src/devices/gridstream.c

```diff
@@ -0,0 +1,323 @@
+            case 0x55:
+                sprintf(destwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[5], b[6], b[7], b[8], b[9], b[10]);
+                sprintf(srcwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[11], b[12], b[13], b[14], b[15], b[16]);
+                sprintf(srcaddress_str, "%02x%02x%02x%02x", b[24], b[25], b[26], b[27]);
+                uptime = ((uint32_t)b[18] << 24) | (b[19] << 16) | (b[20] << 8) | b[21];
+
+                /* clang-format off */
+                        data = data_make(
+                                "model",        "",                 DATA_STRING,    "LandisGyr-GS",
+                                "networkID",    "Network ID",       DATA_STRING,    found_crc,
+                                "location",     "Location",         DATA_STRING,    known_crc_init[crcidx].location,
+                                "provider",     "Provider",         DATA_STRING,    known_crc_init[crcidx].provider,
+                                "id",           "Source Meter ID",  DATA_STRING,    srcaddress_str,
+                                "subtype",      "",                 DATA_INT,       subtype,
+                                "wanaddress",   "",                 DATA_STRING,    srcwanaddress_str,
+                                "destaddress",  "",                 DATA_STRING,    destwanaddress_str,
+                                "uptime",       "",                 DATA_INT,       uptime,
+                                "protoversion",  "",                 DATA_INT,       protocol_version,
+                                "mic",          "Integrity",        DATA_STRING,    "CRC",
+                                NULL);
+                /* clang-format on */
+
+                decoder_output_data(decoder, data);
+                break;
+            case 0xD2:
+
+                /* clang-format off */
+                        data = data_make(
+                                "model",        "",             DATA_STRING, "LandisGyr-GS",
+                                "networkID",    "Network ID",   DATA_STRING, found_crc,
+                                "location",     "Location",     DATA_STRING, known_crc_init[crcidx].location,
+                                "provider",     "Provider",     DATA_STRING, known_crc_init[crcidx].provider,
+                                "subtype",      "",             DATA_INT,    subtype,
+                                "protoversion", "",             DATA_INT,    protocol_version,
+                                "mic",          "Integrity",    DATA_STRING, "CRC",
+                                NULL);
+                /* clang-format on */
+
+                decoder_output_data(decoder, data);
+                break;
+            case 0xD5:
+                sprintf(destaddress_str, "%02x%02x%02x%02x", b[5], b[6], b[7], b[8]);
+                sprintf(srcaddress_str, "%02x%02x%02x%02x", b[9], b[10], b[11], b[12]);
+                if (stream_len == 0x47) {

```

Comment:
There are conditional "include this item if this condition is true" constructs.   That would perhaps make this easier to understand, if that fits the situation.

Reply:
See fineoffset_ws80.c and look for DATA_COND.   There is a way to have what is basically an if statement and if true, the item is added, and if not, it isn't.  At least that's my understanding.